### ISSUE TITLE
Table.hist with bins as np.arange does not follow pyplot's x-labels

### ISSUE THREAD

**Issue by ketozhang** [data-8/datascience#347]
See image below

![image](https://user-images.githubusercontent.com/13506487/53926595-ea9cc680-4038-11e9-88b4-86db483c89be.png)

The main issue comes from floating arithmetic representation:

```python
>>> [x for x in np.arange(0.1,1,0.1)]
[0.1, 0.2, 0.30000000000000004, 0.4, 0.5, 0.6, 0.7000000000000001, 0.8, 0.9]
```

But regardless, pyplot has built in way to fix this in `plt.Axes#hist` which isn't represented by the current datascience package:

```python
# Try this
plt.hist(table.column('value'),  bins=bins)

# or this
fig = plt.figure()
ax = plt.axes()
ax.hist(table.column('value'), bins=bins)
```

![image](https://user-images.githubusercontent.com/13506487/53927408-a5c65f00-403b-11e9-84ad-de2f10e99325.png)

I've scanned the source code but couldn't find the issue. The usage of `bins` in  Table#hist is stored in `vargs` in `plt.Axes#hist(**vargs)`

**Comment by adnanhemani** [comment]
From @papajohn in #311:


Numpy v1.14.x (released Jan 6) changed the algorithm (look for the section titled "Many changes to array printing") for the string representation of floating point numbers. If you add this line of code, it changes back to the 1.13.x algorithm, and the histogram x-axis labels are sane (i.e., don't contain the small error terms):

np.set_printoptions(legacy='1.13')

**Comment by adnanhemani** [comment]
I've made a PR at #381 that should take care of this issue. Please review the change by testing it out locally and making sure that it works as expected now. Thanks!

**Comment by adnanhemani** [comment]
Closed by #381.
